specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-08-02-preview/managedClusters.json

```diff
@@ -3556,9 +3556,87 @@
         "clientId"
       ]
     },
+    "FlexProfile": {
+      "type": "object",
+      "properties": {
+        "scaleProfile": {
+          "$ref": "#/definitions/ScaleProfile",
+          "description": "Specifications on how to scale the AgentPool."
+        }
+      },
+      "description": "Specifications on flex Agentpool."
+    },
+    "ScaleProfile": {
+      "type": "object",
+      "properties": {
+        "manual": {
+          "$ref": "#/definitions/ManualScaleProfile",
+          "description": "Specifications on how to scale the flex agent pool to a fixed size."
+        }
+      },
+      "description": "Specifications on how to scale the AgentPool."
+    },
+    "ManualScaleProfile": {
+      "type": "object",
+      "properties": {
+        "machines": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManualScaleMachineSpec"
+          },
+          "description": "Specifications on how to scale the AgentPool."
+        }
+      }
+    },
+    "ManualScaleMachineSpec": {
+      "type": "object",
+      "properties": {
+        "sizes": {
+          "items": {
+            "type": "string"
+          },
+          "description": "The comma-separated list allowed vm sizes. AKS will use the first available one upon scale."
+        },
+        "count": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Number of nodes (VMs). Allowed values must be in the range of 0 to 1000 (inclusive)."
+        }
+      },
+      "description": "Specifications on how to scale the AgentPool."
+    },
+    "CurrentFlexNodes": {
```

Comment:
Rather than just being directly an array do we want it to be a structure `FlexStatus` (not too fixed on the name), with a property `nodes`? That way it can expand to contain other flex-y status in the future?

I'm not married to this btw just thinking out loud mostly.